### ISSUE TITLE
Removing git_repository in a project resource fails with an inconsistent result after apply error

### ISSUE THREAD

**Issue by ahinh43** [vercel/terraform-provider-vercel#133]
I have a project here with a git repository configured on it:

```hcl
resource "vercel_project" "main" {
  name           = "test"
  framework      = "nextjs"
  
  git_repository = {
    repo = "myteam/my-vercel-repo"
    type = "gitlab"
    production_branch = "main"
  }
}
```

If I apply the configuration then decide I need to remove the `git_repository` field Terraform's plan will propose a update in-place action:

```
 # vercel_project.main will be updated in-place  
~ resource "vercel_project" "main" {
    - git_repository                          = {
        - production_branch = "main" -> null
        - repo              = "myteam/my-vercel-repo" -> null
        - type              = "gitlab" -> null
      } -> null
      id                                      = "prj_..."
      # ...
  }
```

But the apply fails with this error:

```
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to module.vercel_project_temp.vercel_project.main,
│ provider "provider[\"registry.terraform.io/vercel/vercel\"]" produced an
│ unexpected new value: .git_repository: was null, but now
│ cty.ObjectVal(map[string]cty.Value{"production_branch":cty.StringVal("main"),
│ "repo":cty.StringVal("myteam/my-vercel-repo"),
│ "type":cty.StringVal("gitlab")}).
│
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```

And in the console itself nothing ends up happening. Right now if I need to remove the git repository from the project I'd need to do it manually and run `terraform refresh` to sync the change to the state file.

**Comment by dglsparsons** [comment]
Oh no, that sounds like a bug. I'll have a look! Thanks for reporting it.

**Comment by dglsparsons** [comment]
Okay, so the issue seems to be that we support updating the gitrepository, but not removing it. That's a good catch! I'll have a look at getting the unblinking behaviour implemented.

**Comment by dglsparsons** [comment]
With the current behaviour of the provider, we don't support updating the git_repository without re-creating the project (this hasn't been implemented yet, see #105). 

However, terraform incorrectly thinks we can. I've got a fix for this in #134, which is to make changing the `git_repository` field force re-creation of the project. 

This will prevent the error from occurring, at the cost of recreating the project. 

#105 is the longer term solution, but takes a smidge more work.

**Comment by dglsparsons** [comment]
Current workaround would be to taint your resource if you want to update the git_repository, that would force it to be re-created.
